libvips/foreign/heifsave.c

```diff
@@ -580,6 +580,14 @@ vips_foreign_save_heif_build(VipsObject *object)
 		return -1;
 	}
 
+	error = heif_encoder_set_parameter_integer(heif->encoder,
+		"threads", vips_concurrency_get());
```

Comment:
It looks like libheif does not allow more than 16 threads and will fail with the `heif_suberror_Invalid_parameter_value` subcode if you exceed that limit.
https://github.com/strukturag/libheif/blob/c5ea21eb51d8cd14f592cff64f8b179707e45592/libheif/plugins/encoder_aom.cc#L244
https://github.com/strukturag/libheif/blob/c5ea21eb51d8cd14f592cff64f8b179707e45592/libheif/plugins/encoder_rav1e.cc#L122
```suggestion
		"threads", VIPS_MIN(vips_concurrency_get(), 16));
```
(this was noticed on my 12-core/24-thread workstation)

Reply:
Good catch, thank you! Fixed